src/webgpu/api/validation/render_pipeline/createRenderPipeline.spec.ts

```diff
@@ -220,8 +249,204 @@ g.test('color_formats_must_be_renderable')
+  .desc(
+    `If primitive.topology is not "line-strip" or "triangle-strip", primitive.stripIndexFormat must be undefined.`
+  )
+  .params(u =>
+    u
+      .combine('isAsync', [false, true])
+      .combine('topology', [undefined, ...kPrimitiveTopology] as const)
+      .combine('stripIndexFormat', [undefined, ...kIndexFormat] as const)
+  )
+  .fn(async t => {
+    const { isAsync, topology, stripIndexFormat } = t.params;
+
+    const descriptor = t.getDescriptor({ primitive: { topology, stripIndexFormat } });
+
+    const _success =
+      topology === 'line-strip' || topology === 'triangle-strip' || stripIndexFormat === undefined;
+    t.doCreateRenderPipelineTest(isAsync, _success, descriptor);
+  });
+
+g.test('primitive_state,unclipped_depth')
+  .desc(`If primitive.unclippedDepth is true, features must contain "depth-clip-control".`)
+  .params(u => u.combine('isAsync', [false, true]).combine('unclippedDepth', [false, true]))
+  .fn(async t => {
+    const { isAsync, unclippedDepth } = t.params;
+
+    const descriptor = t.getDescriptor({ primitive: { unclippedDepth } });
+
+    const _success = !unclippedDepth || t.device.features.has('depth-clip-control');
+    t.doCreateRenderPipelineTest(isAsync, _success, descriptor);
+  });
+
+g.test('multisample_state,count')
+  .desc(
+    `If multisample.alphaToCoverageEnabled is true, multisample.count must be greater than 1. Sample count must either be 1 or 4.`
+  )
+  .params(u =>
+    u
+      .combine('isAsync', [false, true])
+      .combine('alphaToCoverageEnabled', [false, true])
+      .beginSubcases()
+      .combine('count', [0, 1, 2, 3, 4, 8, 16, 1024])
+  )
+  .fn(async t => {
+    const { isAsync, alphaToCoverageEnabled, count } = t.params;
+
+    const descriptor = t.getDescriptor({ multisample: { count, alphaToCoverageEnabled } });
+
+    const _success = alphaToCoverageEnabled ? count === 4 : count === 1 || count === 4;
+    t.doCreateRenderPipelineTest(isAsync, _success, descriptor);
+  });
+
+g.test('multisample_state,alpha_to_coverage_enabled')
+  .desc(
+    `If sample_mask builtin is a pipeline output of fragment or if multisample.mask is not 0xFFFFFFFF, multisample.alphaToCoverageEnabled should be false.`
+  )
+  .params(u =>
+    u
+      .combine('isAsync', [false, true])
+      .combine('alphaToCoverageEnabled', [false, true])
+      .beginSubcases()
+      .combine('hasSampleMaskOutput', [false, true])
+      .combine('mask', [0, 0x1, 0x2, 0xffffffff])
+  )
+  .fn(async t => {
+    const { isAsync, alphaToCoverageEnabled, mask, hasSampleMaskOutput } = t.params;
+
+    const descriptor = t.getDescriptor({
+      multisample: { mask, alphaToCoverageEnabled, count: 4 },
+      fragmentShaderCode: hasSampleMaskOutput
+        ? `
+      struct Output {
+        @builtin(sample_mask) mask_out: u32,
+        @location(0) color : vec4<f32>,
+      };
+      @fragment fn main() -> Output {
+        var o: Output;
+        o.mask_out = 0x1u;
```

Comment:
uNit: I think make this 0xFFFFFFFFu because that's a "no op" and theoretically an implementation might optimize it out (and we want to make sure it doesn't).